File: convertor.py
# -*- coding: utf-8 -*-
#
#Criado por relue271
#convertor de base decimal para binario, hexadecimal e octal
#
#entrada: 123
#saida:  B:1111011
#        H: 7B
#        O: 173
#
decimal = input("Digite o numero: ")
print("o numero digitado é "+str(decimal))

#binario
aux = decimal
binario = ''
while (aux > 0):
    resto = aux % 2
    aux = aux / 2
    binario += str(resto)
print('B: '),
for k in range((len(binario)-1),-1,-1):
    print(binario[k]),
print





#hexadecimal
aux = decimal
hexadecimal = ''
while (aux > 0):
    resto = aux % 16
    aux = aux /16
    if resto == 10:
        hexadecimal += 'A'
    if resto == 11:
        hexadecimal += 'B'
    if resto == 12:
        hexadecimal += 'C'
    if resto == 13:
        hexadecimal += 'D'
    if resto == 14:
        hexadecimal += 'E'
    if resto == 15:
        hexadecimal += 'F'
    if resto <= 9 and resto >= 0:
        hexadecimal += str(resto)
print('H: '),
for k in range(len(hexadecimal)-1, -1, -1):
    print(hexadecimal[k]),
print





#octal
aux = decimal
octal = ''
while(aux > 0):
    resto = aux % 8
    aux = aux / 8
    octal += str(resto)
print('O: '),
for k in range(len(octal)-1, -1, -1):
    print(octal[k]),
print

File: convertor.pas
Program convertor ;
// Programa que converte um numero decimal em seu binario, hexadecimal e octal respectivo

var 
	decimal : integer;
	// binario
	quociente, i, resto : integer;
	binario : string;
	strresto : string;
	// hexadecimal
	hexa : string;
	// octal
	octal : string;
	

Begin
	// receber um numero decimal
	write('Digite um numero decimal: ');
	readln(decimal);
	writeln();
	
  
	quociente := decimal;
	binario := '';
  // binario
	repeat
		resto := quociente MOD 2;
		quociente := trunc(quociente / 2);
		str(resto:1,strresto);
		binario := concat(strresto,binario);
	until (quociente = 0);
	writeln('binario: ', binario);
	writeln('---');



  // hexadecimal
 	quociente := decimal;
	hexa := '';
	repeat
		resto := quociente MOD 16;
		quociente := trunc(quociente / 16);
		case resto of
			10 : begin
				hexa := concat('A', hexa);
			end;
			11 : begin
				hexa := concat('B', hexa);
			end;
			12 : begin
				hexa := concat('C', hexa);
			end;
			13 : begin
				hexa := concat('D', hexa);
			end;
			14 : begin
				hexa := concat('E', hexa);
			end;
			15 : begin
				hexa := concat('F', hexa);
			end;
			else
				str(resto:1,strresto);
				hexa := concat(strresto,hexa);
		end;
	until (quociente = 0);
	writeln('hexa: ', hexa);
	writeln('---');		



	// octal
  quociente := decimal;
	octal := '';
	repeat
		resto := quociente MOD 8;
		quociente := trunc(quociente / 8);
		str(resto:1,strresto);
		octal := concat(strresto,octal);
	until (quociente = 0);
	writeln('octal: ', octal);
		
End.
